persist/fs/write.go

```diff
@@ -195,33 +235,141 @@ func (w *writer) writeAll(
 			return err
 		}
 	}
-	if _, err := w.indexFdWithDigest.WriteBytes(w.encoder.Bytes()); err != nil {
-		return err
-	}
+
+	w.indexEntries = append(w.indexEntries, entry)
 	w.currIdx++
 
 	return nil
 }
 
-func (w *writer) close() error {
+func (w *writer) writeIndexEntriesSummariesBloomFilter() error {
+	// NB(r): Write the index file in order, in the future we could write
+	// these in order to avoid this sort at the end however that does require
+	// significant changes in the storage/databaseShard to store things in order
+	// which would sacrifice O(1) insertion of new series we currently have.
+	//
+	// Probably do want to do this at the end still however so we don't stripe
+	// writes to two different files during the write loop.
+	sort.Sort(indexEntries(w.indexEntries))
+
+	summariesApprox := float64(len(w.indexEntries)) * w.summariesPercent
+	summaryEvery := 0
+	if summariesApprox > 0 {
+		summaryEvery = int(math.Floor(float64(len(w.indexEntries)) / summariesApprox))
+	}
+
+	// Write the index entries and calculate the bloom filter
+	n, p := uint(w.currIdx), w.bloomFilterFalsePositivePercent
+	m, k := xsets.BloomFilterEstimate(n, p)
+	bloomFilter := xsets.NewBloomFilter(m, k)
+
+	var offset int64
+	for i := range w.indexEntries {
+		id := w.indexEntries[i].id.Data().Get()
+
+		entry := schema.IndexEntry{
```

Comment:
Kind of annoying we have to do this conversion because of a single field difference but I dont know if there is much that can be done about it

Reply:
Yeah, not too big a deal it'll just be a memcopy to a stack allocated piece of memory though.